File: app/src/main/java/com/playdate/app/ui/blockuser/UnblockBottomSheet.java
package com.playdate.app.ui.blockuser;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.RelativeLayout;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.google.android.material.bottomsheet.BottomSheetDialogFragment;
import com.playdate.app.R;
import com.playdate.app.data.api.GetDataService;
import com.playdate.app.data.api.RetrofitClientInstance;
import com.playdate.app.model.LoginResponse;
import com.playdate.app.util.common.TransparentProgressDialog;
import com.playdate.app.util.session.SessionPref;

import java.util.HashMap;
import java.util.Map;

import retrofit2.Call;
import retrofit2.Response;

public class UnblockBottomSheet extends BottomSheetDialogFragment {

    private final String userID;
    private final BlockUserActivity blockUserActivity;

    public UnblockBottomSheet(String userID, BlockUserActivity blockUserActivity) {
        this.userID = userID;
        this.blockUserActivity = blockUserActivity;
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.frag_unblock_bottom_sheet, container, false);
        RelativeLayout rl_unblock = view.findViewById(R.id.rl_unblock);
        rl_unblock.setOnClickListener(v -> callUnBlockUser());

        return view;
    }

    private void callUnBlockUser() {
        SessionPref pref = SessionPref.getInstance(getActivity());

        GetDataService service = RetrofitClientInstance.getRetrofitInstance().create(GetDataService.class);
        Map<String, String> hashMap = new HashMap<>();
        hashMap.put("userId", pref.getStringVal(SessionPref.LoginUserID));
        hashMap.put("action", "Block");//Block or Report
        hashMap.put("toUserId", userID);


        TransparentProgressDialog pd = TransparentProgressDialog.getInstance(getActivity());
        pd.show();

        Call<LoginResponse> call = service.removeUserReportBlock("Bearer " + pref.getStringVal(SessionPref.LoginUsertoken), hashMap);
        call.enqueue(new retrofit2.Callback<LoginResponse>() {
            @Override
            public void onResponse(Call<LoginResponse> call, Response<LoginResponse> response) {
                pd.cancel();
                dismiss();
                if (response.code() == 200) {
                    if (response.body().getStatus() == 1) {
//                        socialFeedAdapter.notifyDataSetChanged();
                        Toast.makeText(getActivity(), "User unblocked successsfully!", Toast.LENGTH_SHORT).show();
                        blockUserActivity.callAPI();
                    } else {
                    }
                } else {

                }


            }

            @Override
            public void onFailure(Call<LoginResponse> call, Throwable t) {
                t.printStackTrace();
                pd.cancel();
//                Toast.makeText(BioActivity.this, "Something went wrong...Please try later!", Toast.LENGTH_SHORT).show();
            }
        });

    }

}

File: app/src/main/java/com/playdate/app/ui/blockuser/UnblockBottomSheet.kt
package com.playdate.app.ui.blockuser

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.RelativeLayout
import android.widget.Toast
import com.google.android.material.bottomsheet.BottomSheetDialogFragment
import com.playdate.app.R
import com.playdate.app.data.api.GetDataService
import com.playdate.app.data.api.RetrofitClientInstance
import com.playdate.app.model.LoginResponse
import com.playdate.app.util.common.TransparentProgressDialog
import com.playdate.app.util.session.SessionPref
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response
import java.util.*

class UnblockBottomSheet(
    private val userID: String,
    private val blockUserActivity: BlockUserActivity
) : BottomSheetDialogFragment() {
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        val view = inflater.inflate(R.layout.frag_unblock_bottom_sheet, container, false)
        val rl_unblock = view.findViewById<RelativeLayout>(R.id.rl_unblock)
        rl_unblock.setOnClickListener { v: View? -> callUnBlockUser() }
        return view
    }

    private fun callUnBlockUser() {
        val pref = SessionPref.getInstance(activity)
        val service = RetrofitClientInstance.getRetrofitInstance().create(
            GetDataService::class.java
        )
        val hashMap: MutableMap<String, String> = HashMap()
        hashMap["userId"] = pref.getStringVal(SessionPref.LoginUserID)
        hashMap["action"] = "Block" //Block or Report
        hashMap["toUserId"] = userID
        val pd = TransparentProgressDialog.getInstance(activity)
        pd.show()
        val call = service.removeUserReportBlock(
            "Bearer " + pref.getStringVal(SessionPref.LoginUsertoken),
            hashMap
        )
        call.enqueue(object : Callback<LoginResponse> {
            override fun onResponse(call: Call<LoginResponse>, response: Response<LoginResponse>) {
                pd.cancel()
                dismiss()
                if (response.code() == 200) {
                    if (response.body()!!.status == 1) {
                        Toast.makeText(
                            activity,
                            "User unblocked successsfully!",
                            Toast.LENGTH_SHORT
                        ).show()
                        blockUserActivity.callAPI()
                    } else {
                    }
                } else {
                }
            }

            override fun onFailure(call: Call<LoginResponse>, t: Throwable) {
                t.printStackTrace()
                pd.cancel()
            }
        })
    }
}